xml-parser/src/main/java/org/uniprot/core/xml/uniparc/UniParcDBCrossReferenceConverter.java

```diff
@@ -57,34 +103,62 @@ public DbReferenceType toXml(UniParcCrossReference uniObj) {
         if (uniObj.getVersion() != null) xmlObj.setVersion(uniObj.getVersion());
         xmlObj.setCreated(XmlConverterHelper.dateToXml(uniObj.getCreated()));
         xmlObj.setLast(XmlConverterHelper.dateToXml(uniObj.getLastUpdated()));
-        uniObj.getProperties().stream()
-                .map(propertyConverter::toXml)
-                .forEach(val -> xmlObj.getProperty().add(val));
+
+        List<PropertyType> properties = new ArrayList<>();
+        if (Utils.notNullNotEmpty(uniObj.getGeneName())) {
+            properties.add(createProperty(PROPERTY_GENE_NAME, uniObj.getGeneName()));
+        }
+        if (Utils.notNullNotEmpty(uniObj.getProteinName())) {
+            properties.add(createProperty(PROPERTY_PROTEIN_NAME, uniObj.getProteinName()));
+        }
+        if (Utils.notNullNotEmpty(uniObj.getChain())) {
+            properties.add(createProperty(PROPERTY_CHAIN, uniObj.getChain()));
+        }
+        if (Utils.notNullNotEmpty(uniObj.getNcbiGi())) {
+            properties.add(createProperty(PROPERTY_NCBI_GI, uniObj.getNcbiGi()));
+        }
+        if (Utils.notNullNotEmpty(uniObj.getProteomeId())) {
+            properties.add(createProperty(PROPERTY_PROTEOME_ID, uniObj.getProteomeId()));
+        }
+        if (Utils.notNullNotEmpty(uniObj.getComponent())) {
+            properties.add(createProperty(PROPERTY_COMPONENT, uniObj.getComponent()));
+        }
+        if (Utils.notNull(uniObj.getTaxonomy())) {
+            String taxonId = String.valueOf(uniObj.getTaxonomy().getTaxonId());
+            properties.add(createProperty(PROPERTY_NCBI_TAXONOMY_ID, taxonId));
+        }
+        xmlObj.getProperty().addAll(properties);
+
         return xmlObj;
     }
 
-    class PropertyConverter implements Converter<PropertyType, Property> {
-        private final ObjectFactory xmlFactory;
-
-        public PropertyConverter() {
-            this(new ObjectFactory());
+    private Organism convertTaxonomy(String taxId) {
+        OrganismBuilder builder = new OrganismBuilder().taxonId(Long.parseLong(taxId));
+        Optional<TaxonomicNode> opNode = getTaxonomyNode(taxId);
+        if (opNode.isPresent()) {
+            TaxonomicNode node = opNode.get();
+            builder.scientificName(node.scientificName());
+            if (!Strings.isNullOrEmpty(node.commonName())) {
```

Comment:
Do we need to null/empty check scientific name?

Reply:
scientificName is a required field for Taxonomy